aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -144,6 +144,27 @@ void float8_copy_kernel_cuda(TensorIteratorBase &iter) {
          gpu_kernel(iter, [] GPU_LAMBDA(Float8_e5m2fnuz x) { return x; });
          break;
     }
+  } else if (dtype == kFloat8_e8m0fnu) {
```

Comment:
this is copy-pasta from above, sounds like a good candidate for clean-up in a separate PR if we understand why the hack was done for other float8 dtypes

Reply:
Is this the actual cast on device code ? e.g. a.to(float32).to(float8)

Reply:
If so maybe add somewhere that we should check if we want to use any intrinsics for faster casts

Reply:
tracking in https://github.com/pytorch/pytorch/issues/146647